lib/kinde_sdk.rb

```
@@ -123,9 +133,31 @@ def client(tokens_hash, auto_refresh_tokens = @config.auto_refresh_tokens, force
+      end
+      
+      parsed = URI.parse(normalized_domain)
+      raise ArgumentError, "invalid domain format: #{domain}" if parsed.host.nil? || parsed.host.empty?
+      
+      scheme = parsed.scheme || 'https'
+      # Only omit port if it matches the default for the scheme
+      default_port = scheme == 'https' ? 443 : 80
+      host_with_port = parsed.port && parsed.port != default_port ? "#{parsed.host}:#{parsed.port}" : parsed.host
+      
+      "#{scheme}://#{host_with_port}/logout#{query ? "?#{query}" : ''}"
     end
```

RuboCop (1.82.1)

[convention] Assignment Branch Condition size for `logout_url` is too high. [<7, 23, 15> 28.34/23]

(Metrics/AbcSize)